modules.d/90overlay-root/overlay-mount.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+# make a read-only nfsroot writeable by using overlayfs
+# the nfsroot is already mounted to $NEWROOT
+# add the parameter rootovl to the kernel, to activate this feature
+
+. /lib/dracut-lib.sh
+
+if ! getargbool 0 rootovl ; then
+    return
+fi
+
+# Move root
+mkdir -p /live/image
+mount --move $NEWROOT /live/image
+
+# Create tmpfs
+mkdir /cow
+mount -t tmpfs -o mode=0755 tmpfs /cow
+mkdir /cow/work /cow/rw
+
+# Merge both to new Filesystem
+mount -t overlay -o lowerdir=/live/image,upperdir=/cow/rw,workdir=/cow/work,default_permissions overlay $NEWROOT
```

Comment:
It might be a performance (and memory) improvement to use `redirect_dir=on,metacopy=on` here to avoid copying of data to the tmpfs. Those options are used from the kernel configuration if unspecified, so depending on the setup that might be the default.